From: Michael Biebl <biebl@debian.org>
Date: Fri, 23 Aug 2019 23:34:45 +0200
Subject: Drop dbus activation stub service

This fixes the following problem:

> At the very end of the boot, just after the first user logs in
> (usually using sddm / X) I get the following messages in my logs:
> Nov 18 07:02:33 samd dbus-daemon[2879]: [session uid=1000 pid=2877] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
> Nov 18 07:02:33 samd dbus-daemon[2879]: [session uid=1000 pid=2877] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1

These messages are caused by the "stub" service files that systemd
installs. It installed them because early versions of systemd activation
required them to exist.

Since dbus 1.11.0, a dbus-daemon that is run with --systemd-activation
automatically assumes that o.fd.systemd1 is an activatable
service. As a result, with a new enough dbus version,
/usr/share/dbus-1/services/org.freedesktop.systemd1.service and
/usr/share/dbus-1/system-services/org.freedesktop.systemd1.service should
become unnecessary, and they can be removed.

dbus 1.11.0 was released 2015-12-02.

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=914015
(cherry picked from commit 07125d24eedb71693b3bf2b1f0730cd01aaac2dd)
---
 README                                    |  4 +---
 src/core/meson.build                      |  2 --
 src/core/org.freedesktop.systemd1.service | 13 -------------
 units/meson.build                         |  3 ---
 4 files changed, 1 insertion(+), 21 deletions(-)
 delete mode 100644 src/core/org.freedesktop.systemd1.service

diff --git a/README b/README
index c652319..dac0d0c 100644
--- a/README
+++ b/README
@@ -173,9 +173,7 @@ REQUIREMENTS:
         dependencies:
 
         util-linux >= v2.27.1 required
-        dbus >= 1.4.0 (strictly speaking optional, but recommended)
-                NOTE: If using dbus < 1.9.18, you should override the default
-                policy directory (--with-dbuspolicydir=/etc/dbus-1/system.d).
+        dbus >= 1.11.0 (strictly speaking optional, but recommended)
         dracut (optional)
         polkit (optional)
 
diff --git a/src/core/meson.build b/src/core/meson.build
index f1df760..ecb4c1f 100644
--- a/src/core/meson.build
+++ b/src/core/meson.build
@@ -194,8 +194,6 @@ endforeach
 
 install_data('org.freedesktop.systemd1.conf',
              install_dir : dbuspolicydir)
-install_data('org.freedesktop.systemd1.service',
-             install_dir : dbussystemservicedir)
 
 policy = configure_file(
         input : 'org.freedesktop.systemd1.policy.in',
diff --git a/src/core/org.freedesktop.systemd1.service b/src/core/org.freedesktop.systemd1.service
deleted file mode 100644
index 8bd7302..0000000
--- a/src/core/org.freedesktop.systemd1.service
+++ /dev/null
@@ -1,13 +0,0 @@
-#  SPDX-License-Identifier: LGPL-2.1+
-#
-#  This file is part of systemd.
-#
-#  systemd is free software; you can redistribute it and/or modify it
-#  under the terms of the GNU Lesser General Public License as published by
-#  the Free Software Foundation; either version 2.1 of the License, or
-#  (at your option) any later version.
-
-[D-BUS Service]
-Name=org.freedesktop.systemd1
-Exec=/bin/false
-User=root
diff --git a/units/meson.build b/units/meson.build
index 52f8b77..f8c9a42 100644
--- a/units/meson.build
+++ b/units/meson.build
@@ -305,9 +305,6 @@ install_data('user-.slice.d/10-defaults.conf',
 meson.add_install_script(meson_make_symlink,
                          join_paths(pkgsysconfdir, 'user'),
                          join_paths(sysconfdir, 'xdg/systemd/user'))
-meson.add_install_script(meson_make_symlink,
-                         join_paths(dbussystemservicedir, 'org.freedesktop.systemd1.service'),
-                         join_paths(dbussessionservicedir, 'org.freedesktop.systemd1.service'))
 if conf.get('HAVE_SYSV_COMPAT') == 1
         foreach i : [1, 2, 3, 4, 5]
                 meson.add_install_script(
